### PR TITLE
Initial port to Dart 2

### PR DESCRIPTION
**Pull request by domesticmouse**
I'm looking at what is required to bring this sample up to date. I've made the changes required to make the compiler and linter happy, but I'm unsure how Harry Terkelsen was testing his code during development. Any hints as to how to proceed gladly accepted...

### CODE REVIEW

**Comment by kanaka** [comment]
With an updated build of the docker image you should be able to test like this from the dart directory:

```
make DOCKERIZE=1 "test^dart"
```

**Comment by kanaka** [comment]
Once that passes locally for you, let me know and I'll update the docker hub image so that Travis will pass.

**Comment by domesticmouse** [comment]
The `Makefile` is unhappy with my macOS environment:

```bash
$ make DOCKERIZE=1 "test^dart"
docker run -it --rm -u 120588 -v /Users/brettmorgan/Documents/GitHub/mal/:/mal -w /mal/dart kanaka/mal-test-dart /Applications/Xcode.app/Contents/Developer/usr/bin/make step0_repl.dart
docker: Error response from daemon: OCI runtime create failed: container_linux.go:346: starting container process caused "exec: \"/Applications/Xcode.app/Contents/Developer/usr/bin/make\": stat /Applications/Xcode.app/Contents/Developer/usr/bin/make: no such file or directory": unknown.
make: *** [dart/step0_repl.dart] Error 127
$ which make
/usr/bin/make
```

Suggestions?

**Comment by kanaka** [comment]
Looks like the make command is setting the MAKE variable on the outside of the container which is getting passed through and used instead the container. Try overriding it like this:

```
make DOCKERIZE=1 MAKE=/usr/bin/make "test^dart"
```

File: dart/.packages [hunk 1]
```diff
@@ -1,2 +1,17 @@
...
+logging:file:///Users/brettmorgan/.pub-cache/hosted/pub.dartlang.org/logging-0.11.3+2/lib/
+meta:file:///Users/brettmorgan/.pub-cache/hosted/pub.dartlang.org/meta-1.1.8/lib/
+path:file:///Users/brettmorgan/.pub-cache/hosted/pub.dartlang.org/path-1.6.4/lib/
+pedantic:file:///Users/brettmorgan/.pub-cache/hosted/pub.dartlang.org/pedantic-1.8.0+1/lib/
+source_span:file:///Users/brettmorgan/.pub-cache/hosted/pub.dartlang.org/source_span-1.5.5/lib/
+string_scanner:file:///Users/brettmorgan/.pub-cache/hosted/pub.dartlang.org/string_scanner-1.0.5/lib/
+term_glyph:file:///Users/brettmorgan/.pub-cache/hosted/pub.dartlang.org/term_glyph-1.1.0/lib/
+tuneup:file:///Users/brettmorgan/.pub-cache/hosted/pub.dartlang.org/tuneup-0.3.6+1/lib/
+yaml:file:///Users/brettmorgan/.pub-cache/hosted/pub.dartlang.org/yaml-2.2.0/lib/
 mal:lib/
```
Comment (kanaka) [inline dart/.packages]: Please drop this change from the push. It contains local file paths.

Comment (domesticmouse) [reply dart/.packages]: done

**Pull request closed without merging**